### DISCUSSION TITLE
Impressum Section

### DISCUSSION THREAD

**Discussion by PrXmE** [filipecarneiro/ELabel#15] [Ideas]
Considering the strict legal standards in Europe, particularly in Germany, having an Impressum section is essential to ensure compliance and transparency, meeting the necessary legal obligations while fostering user trust.

**Comment by filipecarneiro** [comment]
Hi,

Thank you for your suggestion.

As I understood, Impressum is legally required in 3 countries (Germany, Austria, and Switzerland) and only for commercial websites. As these pages doesn’t generate money, an Impressum is not required. Do you agree?

Actually, e-label pages are restricted, by the UE law, to have a link to a commercial website or any type of commercial and marketing information.

Saying that, the e-label page is also required to identify the responsible of the information provided, with its name and address ("bottled by" or “imported by”).
 
Maybe adding an optional contact information and VAT number, as required in the [Art. 5 TMG](https://dejure.org/gesetze/TMG/5.html), will satisfy your need. What do you think?

**Comment by filipecarneiro** [comment]
I'll have to ask if including the contact information, like an email or telephone number, will violate Article 119 (5):

> 5.
> By way of derogation from paragraph 1, point (i), the list of ingredients may be provided by electronic means identified on the package or on a label attached thereto. In such cases, the following requirements apply:
> (a) no user data shall be collected or tracked;
> (b) the list of ingredients shall not be displayed with other information intended for sales or marketing purposes; and
> (c) the indication of the particulars referred to in Article 9(1), point (c), of Regulation (EU) No 1169/2011 shall appear directly on the package or on a label attached thereto.

I’m sure that a link to the responsible website is not allowed:

> Is it possible to link, via a QR Code or similar, the ‘electronic label’ presenting the full nutrition declaration and list of ingredients to the homepage of the producer as a part of its website?
>
> No. Article 119(5) of the amended CMO Regulation, provides that the information on the full nutritional declaration
> and list of ingredients shall not be displayed with other information intended for sales or marketing purposes, and
> that no user data shall be collected or tracked. In the view of the Commission services, the presentation of this
> compulsory information as part of the producers’ website does not seem to comply with the conditions set out in
> 
> Article 119(5), as the website of a wine producer is typically expected to contain commercial information relevant
> for marketing and/or sales. In addition, websites normally track information on the users.

([C/2023/1190](https://eur-lex.europa.eu/legal-content/EN/TXT/PDF/?uri=OJ:C_202301190) Questions and answers on the implementation of new EU wine labelling provisions)

**Comment by susedv** [comment]
I'd like to take part in this discussion. I'm not sure about the legal stuff, but I have checked some commercial e-label providers like de.e-label.io, e-label.online and some other and I see that their demo e-labels include imprint at the bottom (address and VAT sometimes).

**Reply by filipecarneiro** [reply]
Address is already available. I'll add VAT.

**Comment by susedv** [comment]
@filipecarneiro

What do you think about adding a field where the user can enter a custom footer? It may be an imprint or it may be any other kind of information, the user is resposible itself for insuring that this appended content meets the law.

**Reply by filipecarneiro** [reply]
Good idea! I'll do that.

**Comment by susedv** [comment]
Our customer requested both contents: Imprint and Privacy Policy. Because it would be too much text for the E-Label page, I think they should be put into separate pages and linked in the footer of the E-Label page, like on any other web page. Now I'm thinking about extending E-Label myself, adding a new database Model like class Page { public long ID { get; set; } public string Title { get; set; } public string Content { get; set; } } with the corresponding controller and views and a WYSIWYG editor like CKEditor for the HTML input. The bottom menu in the footer of E-Label will then be generated automatically. What do you think?

**Comment by filipecarneiro** [comment]
I don't want to go down that road of having multiple pages. Then you will have a kind of menu, header, footer, multi-language support on those pages, ... it would be better to just have WordPress with a custom post type!

I would chose a link to a PDF or a custom static HTML page. Note that labels are not allowed to have a link to a commercial website!

**Comment by filipecarneiro** [comment]
**Food Business Operator** now has an `Additional information` field to write a code, VAT number or additional Impressum information.
If you want,  I can also add a link (actually is already coded but I end up not using it, due to legal concerns)

**Reply by susedv** [reply]
I have tested your latest commits. `Additional information` is only single line and doesn't support HTML tags, but imprint information is multiline and contains usually a disclaimer, therefore it cannot be entered there.

There may be multiple legal contents required: Imprint, Disclaimer and Privacy Policy and it may be a lot of text and should be accessible from any page (Login page for example). Therefore I think the right way to do it is to put them into separate pages and link them in the footer.

In my "workaround" branch I added 2 static files "imprint.html" and "privacy-policy.html" inside "wwwroot" and linked them as follows:

![grafik](https://github.com/filipecarneiro/ELabel/assets/43996715/9533ebe5-bc11-40bd-a87f-6514a9452a48)
